security/rbac/engine/engine_test.go

```diff
@@ -0,0 +1,47 @@
+// +build go1.10
+
+/*
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package engine
+
+import (
+	"strings"
+	"testing"
+
+	pb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2"
+)
+
+func TestTooFewRbacs(t *testing.T) {
```

Comment:
I would suggest the following for the tests:
1. use table-driven testing in Go(https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) to combine these three into one test(e.g. TestCelEnigineCreation).
2. tests should cover all the functions you added, and ideally all the executed path(edges of control flow graph) be tested(it's fine if some corner cases are too hard to test, but we at least need to test all the added tests)
3. Given the situation that function
`func evaluateCondition(condition *cel.Program, evalMap map[string]interface{}) (bool, error)`
is not implemented yet, we don't even have control over it. It would be very difficult to mock this function. I would suggest:

1. remove evaluateCondition function and use program.Eval(...) to replace it
2. use different implementations of cel.Program to mock the behavior. In our tests, we can have our own implementation of cel.Program, and set it to celEngine
3. in the mock implementation of cel.Program, we can write some simple code, e.g. if the interface is a map which contains key named "good:, always return a successful evaluation, etc. Note that the purpose is still to test the behaviors in your code, so no need for complex implementations.
LMK if you have any questions, and I am happy to discuss offline.

Reply:
Should I include the tests in this PR? Currently I've been writing unit tests on a different branch.

Reply:
Yes, we usually include implementations and tests in one PR. Unit tests will help identify some bugs in your code, so it's good to have them submitted together.